Adds feature-maintainer, emeritus and updates removal rules (#508)
dapr/community: community-membership.md

- Expected to be responsive to review requests (inactivity for more than 1 month may result in suspension until active again)
- Mentor contributors and reviewers
- May approve code contributions for acceptance
- Inactivity for more than 3 months leads to a removal vote by maintainers and not an automatic removal

## Feature-maintainer

> Feature-maintainers are the technical authority for a feature or specialized area of the project/repository in the Dapr org. They *MUST* have demonstrated both good judgement and responsibility towards the health of that space. Feature Maintainers can set technical direction and make or approve design decisions for their designated space.

Reviewer: ```suggestion
Feature-maintainers are the technical authority for a feature or specialized area of the project/repository in the Dapr org. They must have demonstrated both good judgement and responsibility towards the health of that space. Feature Maintainers can set technical direction and make or approve design decisions for their designated space.
```

> Feature-maintainers have binding vote only in proposals and changes that affect their designated space.

Reviewer: ```suggestion
Feature-maintainers have binding vote only in proposals and changes that affect their feature area.
```

- Make and approve technical design decisions for the area
- Set technical direction and priorities for the area.
- Define milestones and releases working along with the maintainers
- Decides on when PRs are merged to control the release scope
- Mentor and guide approvers and contributors of the area
- Escalate approver and maintainer workflow concerns. (For example responsiveness, availability, and general contributor community health) to the STC
- Ensure continued health of area:
  - Adequate test coverage to confidently release
  - Tests are passing reliably (not flaky) and are fixed when they fail.
  - Work with other feature-maintainers & maintainers to maintain the project's overall health and success holistically.
- Membership tracked in feature-maintainer.md entry and scoped to a area.
- Must maintain components, review, and approve proposals for enhancing areas falling under the user area.
- Actively participate in issue triages and PR reviews
- Set milestone priorities for the areas working with the repo maintainers.
- Ensure a healthy process for discussion and decision making is in place

## Maintainer

> Maintainers are the technical authority for a repository in the Dapr org. They *MUST* have demonstrated both good judgement and responsibility towards the health of that repository. Maintainers *MUST* set technical direction and make or approve design decisions for their repository - either directly or through delegation of these responsibilities to approvers or feature maintainers.

Reviewer: ```suggestion
Maintainers are the technical authority for a repository in the Dapr org. They must have demonstrated both good judgement and responsibility towards the health of that repository. Maintainers must set technical direction and make or approve design decisions for their repository - either directly or through delegation of these responsibilities to approvers or feature maintainers.
```

- Ensure a healthy process for discussion and decision making is in place
- Work with other maintainers to maintain the project's overall health and success holistically

> Maintainers *MUST* remain active. If they are unresponsive for >3 months, a maintainer might start a removal voting process, requiring a super-majority of the other maintainers of the given repository or STC to approve removal.

Reviewer: ```suggestion
Maintainers must remain active. If they are unresponsive for >3 months, a maintainer might start a removal voting process, requiring a super-majority of the other maintainers of the given repository or STC to approve removal.
```
